src/sdk/noobaa_s3_client/noobaa_s3_client.js

```diff
@@ -89,7 +97,7 @@ function get_requestHandler_with_suitable_agent(endpoint) {
             httpAgent: agent
         });
     } else {
-        throw new Error(`Protocol of ${endpoint} doesn't have a matching agent (valid only for HTTP, HTTPS, WSS only)`);
+        dbg.error(`Protocol of ${endpoint} doesn't have a matching agent (valid only for HTTP, HTTPS, WSS only)`);
```

Comment:
why did you remove the `throw`, is this a valid state? if yes then we shouldn't log an error. if it's an invalid state it's better to throw an identify it earlier

Reply:
I removed it because this function meant to get `NodeHttpHandler` and I thought that there might be cases that a S3 client doesn't really need this property and it might be fine, but rethinking it - in the cases where it is needed and someone used it improperly it would create weird errors... I'll throw the error as it was.